trace/kafka/kafka.go

```diff
@@ -123,21 +127,33 @@ func (ap *AsyncProducer) propagateError() {
 	}
 }
 
-func createProducerMessage(ctx context.Context, msg *Message, sp opentracing.Span) (*sarama.ProducerMessage, error) {
+func createProducerMessage(ctx context.Context, msg *Message, sp opentracing.Span, enc encoding.EncodeFunc, ct string) (*sarama.ProducerMessage, error) {
 	c := kafkaHeadersCarrier{}
 	err := sp.Tracer().Inject(sp.Context(), opentracing.TextMap, &c)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to inject tracing headers")
 	}
-	var saramaKey sarama.Encoder
+
+	if len(ct) != 0 {
```

Comment:
Since we will always setting both the conten-type and the encoder we should set this always.
BTW The most common way to check for a empty string in Go is the comparisson with "". Using len() someone might think that it is a slice of something...